### ISSUE TITLE
Fix references on rename and move

### ISSUE THREAD

**Issue by cseickel** [nvim-neo-tree/neo-tree.nvim#47]
Use the LSP when available to fix references after (or before?) a file is moved or renamed.

https://github.com/jose-elias-alvarez/nvim-lsp-ts-utils/blob/main/lua/nvim-lsp-ts-utils/rename-file.lua

**Comment by gegoune** [comment]
Is that for updating buffer after corresponding buffer has been renamed/moved?

**Comment by cseickel** [comment]
> Is that for updating buffer after corresponding buffer has been renamed/moved?

No, I think I actually have that already. This means that if the given file is imported into other files in your project, also fix those import statements to point to the new name.

**Comment by nyngwang** [comment]
To @cseickel, @gegoune:

> [...] as renaming a.lua to b.lua leaves me with open buffer named `a.lua`, ([...]) and no `b.lua` gets opened. [...]

I have the same result.

**Comment by cseickel** [comment]
Sorry I never responded to this. It seems like I remembered wrong and rename from neo-tree does not clean up buffers. This should probably be a separate issue.

**Comment by Gelio** [comment]
I've just noticed I posted a comment in the wrong issue :sweat_smile: My bad. I'll copy the comment from #308 as it is more relevant here.

---

FYI I was able to implement this functionality for `tsserver`. I added the following handler https://github.com/Gelio/ubuntu-dotfiles/blob/d5f693db0e5afd10efacb3269f3a96bf3bade834/install/neovim/stowed/.config/nvim/lua/plugins.lua#L116-L134 for the rename and move neo-tree events https://github.com/Gelio/ubuntu-dotfiles/blob/d5f693db0e5afd10efacb3269f3a96bf3bade834/install/neovim/stowed/.config/nvim/lua/plugins.lua#L176-L183

It may be worth adding this snippet to the documentation or the codebase so others can easily copy/require it and use it themselves.

With this snippet, renaming/moving a file or directory in neo-tree automatically updates import paths in TypeScript.

Keep in mind that the `tsserver` language server needs to be running in the background for this command to run. This is easy enough, though. Just open any TypeScript file and then your subsequent renames/moves will automatically update imports in other files.
